code/game/machinery/computer/robot.dm

```diff
@@ -242,15 +243,23 @@
 				if(R.scrambledcodes)
 					return
 				// whatever weirdness this is supposed to be, but that is how the href gets added, so here it is again
-				if(istype(R) && istype(usr, /mob/living/silicon) && usr.mind.special_role && R.emagged != 1)
+				if(istype(R, /mob/living/silicon/robot) && istype(usr, /mob/living/silicon/ai) && ismalf(usr) && R.emagged != 1)
```

Comment:
I think this was supposed to let traitor cyborgs hack themselves, though I'm not sure. Your change would prevent that. cc @jknpj

Reply:
Yes, traitor borgs are supposed to be able to emag themselves.

```suggestion
				if(isrobot(R) && issilicon(usr) && (ismalf(usr) || (usr == R && istraitor(usr)) && !R.emagged)
```

Maybe change it to be something like this?

Reply:
that should work, I'll test it soon. I never knew that was a feature.